ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -472,47 +476,64 @@ Tensor convert_python_tensors_to_tt_tensors(
 
 template <typename T>
 owned_buffer::Buffer<T> create_row_major_owned_buffer(
-    owned_buffer::Buffer<T> owned_buffer, const ttnn::TensorSpec& tensor_spec) {
-    if (tensor_spec.layout() == Layout::TILE) {
-        auto data = tensor_impl::convert_layout_tile_to_row_major(
-            tensor_spec.physical_shape(), tensor_spec.tile(), owned_buffer);
-        return owned_buffer::create(std::move(data));
+    owned_buffer::Buffer<T> owned_buffer, const ttnn::TensorSpec& tensor_spec, const bool legacy_output) {
+    if (tensor_spec.memory_config().is_sharded()) {
+        TT_FATAL(
+            tensor_spec.memory_config().shard_spec.has_value(),
+            "Sharded tensors must have a shard spec when converting from tt tensors!");
+    }
```

Comment:
We are calling this when converting back from ttnn tensor... So is it ok to assume `tensor_spec` is correctly created already?

Reply:
Yes

Reply:
Ok to remove the assertion then?

Reply:
Technically, you can do weird things with output tensors. It's safer to have the assert both ways (unless we make shard spec read only or something).